astropy/table/operations.py

```diff
@@ -636,8 +636,14 @@ def _join(left, right, keys=None, join_type='inner',
         # If the output table is masked then set the output column masking
         # accordingly.  Check for columns that don't support a mask attribute.
         if masked:
+            # array_mask is 1-d corresponding to length of output column.  We need
+            # make it have the correct shape for broadcasting, ie. (length, 1, 1, ..).
+            # Mixin columns might not have ndim attribute so use len(col.shape).
+            mask_shape = [out[out_name].shape[0]] + [1] * (len(out[out_name].shape) - 1)
```

Comment:
@taldcroft - here, one could just write
```
array_mask.shape = (out[out_name].shape[0],) + (1,) * (len(out[out_name].shape) - 1)
```
This is one line shorter and slightly faster and I think just as clear about intent.  Also, I always think of the shape as a tuple, so am using that; it also happens to be substantially faster than lists (though likely in the noise here):
```
In [2]: %timeit (10,) + (5-1)*(1,)
100000000 loops, best of 3: 19.6 ns per loop

In [3]: %timeit [10,] + (5-1)*[1]
1000000 loops, best of 3: 645 ns per loop
```